### ISSUE TITLE
Mypy doesn't show errors when base method has `*args: Any` or `**kwargs: Any`

### ISSUE THREAD

**Issue by denisshaf** [python/mypy#18980]
**Bug Report**

If the base class / prototype has a method that accepts only `*args: Any` or / and `**kwargs: Any`, mypy doesn't show any errors, even if the child class accepts only limited set of parameters. If I add `a: int` to `foo` in the base class, nothing changes. If I add `b: int` to `foo` in the base class, mypy show an error "Signature of "foo" incompatible with supertype "HasFoo"", that is expected.

**To Reproduce**

```python
from typing import Protocol, Any


class HasFoo:
    def foo(self, *arg: Any, **kwargs: Any) -> str: return 'foo'


class Foo(HasFoo):
    def foo(self, a: int) -> str:
        print(a)
        return 'foo'
        

f: HasFoo = Foo()
f.foo(3, 4, 5)
```

Gist URL: https://gist.github.com/mypy-play/1a858925c056f8b5391e09a247b29ece
Playground URL: https://mypy-play.net/?mypy=latest&python=3.13&gist=1a858925c056f8b5391e09a247b29ece

**Expected Behavior**

Mypy should raise an error "Signature of "foo" incompatible with supertype "HasFoo"". 

**Actual Behavior**

Mypy doesn't raise any errors. The code above fails in runtime.

**Your Environment**

- Mypy version used: 1.15.0
- Mypy command-line flags: -
- Mypy configuration options from `mypy.ini` (and other config files):
```toml
[tool.mypy]
disallow_untyped_defs = true
disallow_any_unimported = true
no_implicit_optional = true
check_untyped_defs = true
warn_return_any = true
warn_unused_ignores = true
packages = ['src']
```
- Python version used: 3.13.3

**Comment by TeamSpen210** [comment]
This is intentional behaviour. The combination of `*args: Any` and `**kwargs: Any` [is special](https://typing.python.org/en/latest/spec/callables.html#meaning-of-in-callable), allowing any possible signature.

**Comment by A5rocks** [comment]
And `*args: Any` by itself is also intentionally special. (Due to type var tuples)
